Steve/aix

stevenkboyd opened the pull request:
This is a duplicate implementation of aix as to what we have in the bonelab repository. However, it's become difficult to manage large numbers of files using the python implementation, and also for FE workflow requires both n88 and bonelab repositories to be used. This implementation of aix is just for convenience. I couldn't figure out how to write a test with an application, but if I did, the test would confirm the following:

$ ./aix data/test_short_v2.aim -m
"CAMOS_0709" "984" "4818" "TL"

and 

$ ./aix data/test_bincmp_v2.aim -m
"not_found" "not_found" "not_found" "not_found"

Besler commented on CMakeLists.txt:
```diff
@@ -65,6 +66,15 @@ target_link_libraries (AimIO
+		add_executable (aix source/aix.cxx)
+		target_link_libraries (aix AimIO)
+		if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+		    target_link_libraries (AimIO dl pthread)
```
I believe you have the wrong target here. Should it not be `aix`?
```suggestion
		    target_link_libraries (aix dl pthread)
```

stevenkboyd commented:
Oops. I don’t know how I missed that. Thanks for finding and fixing.

By the way, if you look in aix.cxx you’ll see I commented out line 140 because I couldn’t figure out how to adjust the AimFile class in AimIO.cxx/.h so that there was a public method for accessing the byte offset to the start of data (this->block_list[2].offset). If you see how to do this, please let me know as I banged my head against the wall for a while with no success.

When do we merge with the main branch?

Cheers,
Steve


On Apr 2, 2020, at 2:03 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com>> wrote:



@Besler commented on this pull request.

Great idea and implementation! There is a CMake error which I will push corrections for shortly. Hopefully they resolve the issue.

________________________________

In CMakeLists.txt<https://github.com/Numerics88/AimIO/pull/4#discussion_r402575475>:

> @@ -17,6 +17,7 @@ set (CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY ON)
 # === Source code files

 set (SRC
+  source/aix.cxx


Since aix.cxx is a program (has a main function, etc.) it shouldn't be added to the list of sources for the AimIO library.

________________________________

In CMakeLists.txt<https://github.com/Numerics88/AimIO/pull/4#discussion_r402578020>:

> @@ -65,6 +66,15 @@ target_link_libraries (AimIO
                Boost::filesystem Boost::system
 )

+option (N88_BUILD_AIX "Build aix tool." ON)
+if (N88_BUILD_AIX)
+               add_executable (aix source/aix.cxx)
+               target_link_libraries (aix AimIO)
+               if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+                   target_link_libraries (AimIO dl pthread)


I believe you have the wrong target here. Should it not be aix?

⬇️ Suggested change

-                   target_link_libraries (AimIO dl pthread)
+                   target_link_libraries (aix dl pthread)


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#pullrequestreview-386751160>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITVTAEJST75ZEPZ3CYLRKTVSJANCNFSM4L2XGCHA>.

Besler commented:
There are some build errors with boost that I'm working through right now. Let me fix that and give the offset issue a try and then we can merge.

Besler commented:
Okay, build working again. Last thing  to do would be to grab `"!%  Image Data starts at byte offset " << this->block_list[2].offset` as you want. This isn't exposed right now, so we could add a public variable called `byte_offset` and set it when the header is set.
```c++
int         byte_offset;
```

Thoughts?

stevenkboyd commented:
Yes, that’s exactly what I had tried to do, but failed. If you make it work I’ll be very interested in how…

STeve

--
Steven Boyd
PhD, Professor, University of Calgary
Director, McCaig Institute for Bone and Joint Health
Full Signature<http://people.ucalgary.ca/~skboyd/Steven_Boyd/Signature.html>

On Apr 2, 2020, at 3:40 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com>> wrote:



Okay, build working again. Last thing to do would be to grab "!% Image Data starts at byte offset " << this->block_list[2].offset as you want. This isn't exposed right now, so we could add a public variable called byte_offset and set it when the header is set.

int         byte_offset;

Thoughts?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608104713>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITQQYYTPIRGHQAEV7B3RKUA4RANCNFSM4L2XGCHA>.

Besler commented:
Okay, had a mixup on master, rebased master into this branch so once we merge, everything is fixed.

Besler commented:
@stevenkboyd We're good to go! You should check that the aix number I added makes sense. Otherwise, hit merge (bottom of the PR on GitHub) and delete the branch after the merge goes through.

I had written some tests for the `aix` output, but I ran into some `popen` issues with Windows. Not worth the 4 hours to short that out.

stevenkboyd commented:
Hi,

My build isn’t working. Did a file not get committed to the steve/aix branch? See snippet below.

Also, curious why in AimIO.h it can be declared as an “int” instead of “size_t”, which is how the struct is built.

Steve

(faim-devel) $ make
[ 11%] Building CXX object CMakeFiles/AimIO.dir/source/AimIO.cxx.o
[ 22%] Building CXX object CMakeFiles/AimIO.dir/source/Compression.cxx.o
[ 33%] Linking CXX static library libAimIO.a
[ 33%] Built target AimIO
[ 44%] Building CXX object CMakeFiles/aix.dir/source/aix.cxx.o
[ 55%] Linking CXX executable aix
[ 55%] Built target aix
[ 66%] Generating /Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio-build/data/aimio_testdata.tar.gz
-- Found object: "/Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio-build/ExternalData/Objects/MD5/c2d1f2278c5b735c765bd46b8c6a2ee4"
[ 66%] Built target fetch_data_tarball
[ 77%] Generating ../data/test_bincmp_v2.aim
x test_bincmp_v2.aim
x test_bincmp_v3.aim
x test_charcmp_v2.aim
x test_charcmp_v3.aim
x test_short_offset_v2.aim
x test_short_offset_v3.aim
x test_short_v2.aim
x test_short_v3.aim
[ 77%] Built target test_data
[ 88%] Building CXX object tests/CMakeFiles/AimIOTests.dir/AimIOTests.cxx.o
/Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio/tests/AimIOTests.cxx:161:27: error:
      no member named 'byte_offset' in 'AimIO::AimFile'
  ASSERT_EQ (1757, reader.byte_offset);
                   ~~~~~~ ^
/Users/skboyd/Desktop/faim-devel/build/googletest/src/googletest/include/gtest/gtest.h:2011:54: note:
      expanded from macro 'ASSERT_EQ'
# define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
                                                     ^~~~
/Users/skboyd/Desktop/faim-devel/build/googletest/src/googletest/include/gtest/gtest.h:1995:33: note:
      expanded from macro 'GTEST_ASSERT_EQ'
                      expected, actual)
                                ^~~~~~

--
Steven Boyd
PhD, Professor, University of Calgary
Director, McCaig Institute for Bone and Joint Health
Full Signature<http://people.ucalgary.ca/~skboyd/Steven_Boyd/Signature.html>

On Apr 2, 2020, at 7:19 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com>> wrote:



@stevenkboyd<https://github.com/stevenkboyd> We're good to go! You should check that the aix number I added makes sense. Otherwise, hit merge (bottom of the PR on GitHub) and delete the branch after the merge goes through.

I had written some tests for the aix output, but I ran into some popen issues with Windows. Not worth the 4 hours to short that out.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608176971>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITXAYL4DHSNMQS52YSDRKU2RJANCNFSM4L2XGCHA>.

Besler commented:
We can do size_t or int. It's not part of the file definition so we don't read/write like we do the other parts.

As for this issue, you have to delete the build directory and start again. CMAKE files aren't robust in this build.
________________________________
From: stevenkboyd <notifications@github.com>
Sent: Thursday, April 2, 2020 8:35:38 PM
To: Numerics88/AimIO <AimIO@noreply.github.com>
Cc: Bryce Albert Besler <babesler@ucalgary.ca>; Mention <mention@noreply.github.com>
Subject: Re: [Numerics88/AimIO] Steve/aix (#4)


Hi,

My build isn’t working. Did a file not get committed to the steve/aix branch? See snippet below.

Also, curious why in AimIO.h it can be declared as an “int” instead of “size_t”, which is how the struct is built.

Steve

(faim-devel) $ make
[ 11%] Building CXX object CMakeFiles/AimIO.dir/source/AimIO.cxx.o
[ 22%] Building CXX object CMakeFiles/AimIO.dir/source/Compression.cxx.o
[ 33%] Linking CXX static library libAimIO.a
[ 33%] Built target AimIO
[ 44%] Building CXX object CMakeFiles/aix.dir/source/aix.cxx.o
[ 55%] Linking CXX executable aix
[ 55%] Built target aix
[ 66%] Generating /Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio-build/data/aimio_testdata.tar.gz
-- Found object: "/Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio-build/ExternalData/Objects/MD5/c2d1f2278c5b735c765bd46b8c6a2ee4"
[ 66%] Built target fetch_data_tarball
[ 77%] Generating ../data/test_bincmp_v2.aim
x test_bincmp_v2.aim
x test_bincmp_v3.aim
x test_charcmp_v2.aim
x test_charcmp_v3.aim
x test_short_offset_v2.aim
x test_short_offset_v3.aim
x test_short_v2.aim
x test_short_v3.aim
[ 77%] Built target test_data
[ 88%] Building CXX object tests/CMakeFiles/AimIOTests.dir/AimIOTests.cxx.o
/Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio/tests/AimIOTests.cxx:161:27: error:
no member named 'byte_offset' in 'AimIO::AimFile'
ASSERT_EQ (1757, reader.byte_offset);
~~~~~~ ^
/Users/skboyd/Desktop/faim-devel/build/googletest/src/googletest/include/gtest/gtest.h:2011:54: note:
expanded from macro 'ASSERT_EQ'
# define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
^~~~
/Users/skboyd/Desktop/faim-devel/build/googletest/src/googletest/include/gtest/gtest.h:1995:33: note:
expanded from macro 'GTEST_ASSERT_EQ'
expected, actual)
^~~~~~

--
Steven Boyd
PhD, Professor, University of Calgary
Director, McCaig Institute for Bone and Joint Health
Full Signature<http://people.ucalgary.ca/~skboyd/Steven_Boyd/Signature.html>

On Apr 2, 2020, at 7:19 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com>> wrote:



@stevenkboyd<https://github.com/stevenkboyd> We're good to go! You should check that the aix number I added makes sense. Otherwise, hit merge (bottom of the PR on GitHub) and delete the branch after the merge goes through.

I had written some tests for the aix output, but I ran into some popen issues with Windows. Not worth the 4 hours to short that out.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608176971>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITXAYL4DHSNMQS52YSDRKU2RJANCNFSM4L2XGCHA>.


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608196730>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AESNJJ3VQ3VF6X3XWI3TWNDRKVDPVANCNFSM4L2XGCHA>.

stevenkboyd commented:
Hi,

I just deleted and built again, but got same error. Are you able to delete and build from the steve/aix branch?

Steve

--
Steven Boyd
PhD, Professor, University of Calgary
Director, McCaig Institute for Bone and Joint Health
Full Signature<http://people.ucalgary.ca/~skboyd/Steven_Boyd/Signature.html>

On Apr 2, 2020, at 8:37 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com>> wrote:


We can do size_t or int. It's not part of the file definition so we don't read/write like we do the other parts.

As for this issue, you have to delete the build directory and start again. CMAKE files aren't robust in this build.
________________________________
From: stevenkboyd <notifications@github.com<mailto:notifications@github.com>>
Sent: Thursday, April 2, 2020 8:35:38 PM
To: Numerics88/AimIO <AimIO@noreply.github.com<mailto:AimIO@noreply.github.com>>
Cc: Bryce Albert Besler <babesler@ucalgary.ca<mailto:babesler@ucalgary.ca>>; Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [Numerics88/AimIO] Steve/aix (#4)


Hi,

My build isn’t working. Did a file not get committed to the steve/aix branch? See snippet below.

Also, curious why in AimIO.h it can be declared as an “int” instead of “size_t”, which is how the struct is built.

Steve

(faim-devel) $ make
[ 11%] Building CXX object CMakeFiles/AimIO.dir/source/AimIO.cxx.o
[ 22%] Building CXX object CMakeFiles/AimIO.dir/source/Compression.cxx.o
[ 33%] Linking CXX static library libAimIO.a
[ 33%] Built target AimIO
[ 44%] Building CXX object CMakeFiles/aix.dir/source/aix.cxx.o
[ 55%] Linking CXX executable aix
[ 55%] Built target aix
[ 66%] Generating /Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio-build/data/aimio_testdata.tar.gz
-- Found object: "/Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio-build/ExternalData/Objects/MD5/c2d1f2278c5b735c765bd46b8c6a2ee4"
[ 66%] Built target fetch_data_tarball
[ 77%] Generating ../data/test_bincmp_v2.aim
x test_bincmp_v2.aim
x test_bincmp_v3.aim
x test_charcmp_v2.aim
x test_charcmp_v3.aim
x test_short_offset_v2.aim
x test_short_offset_v3.aim
x test_short_v2.aim
x test_short_v3.aim
[ 77%] Built target test_data
[ 88%] Building CXX object tests/CMakeFiles/AimIOTests.dir/AimIOTests.cxx.o
/Users/skboyd/Desktop/faim-devel/build/aimio/src/aimio/tests/AimIOTests.cxx:161:27: error:
no member named 'byte_offset' in 'AimIO::AimFile'
ASSERT_EQ (1757, reader.byte_offset);
~~~~~~ ^
/Users/skboyd/Desktop/faim-devel/build/googletest/src/googletest/include/gtest/gtest.h:2011:54: note:
expanded from macro 'ASSERT_EQ'
# define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
^~~~
/Users/skboyd/Desktop/faim-devel/build/googletest/src/googletest/include/gtest/gtest.h:1995:33: note:
expanded from macro 'GTEST_ASSERT_EQ'
expected, actual)
^~~~~~

--
Steven Boyd
PhD, Professor, University of Calgary
Director, McCaig Institute for Bone and Joint Health
Full Signature<http://people.ucalgary.ca/~skboyd/Steven_Boyd/Signature.html>

On Apr 2, 2020, at 7:19 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:



@stevenkboyd<https://github.com/stevenkboyd> We're good to go! You should check that the aix number I added makes sense. Otherwise, hit merge (bottom of the PR on GitHub) and delete the branch after the merge goes through.

I had written some tests for the aix output, but I ran into some popen issues with Windows. Not worth the 4 hours to short that out.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608176971>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITXAYL4DHSNMQS52YSDRKU2RJANCNFSM4L2XGCHA>.


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608196730>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AESNJJ3VQ3VF6X3XWI3TWNDRKVDPVANCNFSM4L2XGCHA>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608197341>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITWY65YHUN5QJS6S2PDRKVDYBANCNFSM4L2XGCHA>.

Besler commented:
It was working and now it isn't. I think it is an issue with how the sources are setup. I'll give it a look here.

stevenkboyd commented:
Ok, I’m glad to know that I’m not crazy. I recall when I was trying to set up the byte_offset (ironically I chose exactly the same variable name) I could never get it to recognize it as a member of the class.

I had tried to set up a method called GetByteOffset, but the way you did it is more elegant.

Steve

--
Steven Boyd
PhD, Professor, University of Calgary
Director, McCaig Institute for Bone and Joint Health
Full Signature<http://people.ucalgary.ca/~skboyd/Steven_Boyd/Signature.html>

On Apr 2, 2020, at 9:32 PM, Bryce Besler <notifications@github.com<mailto:notifications@github.com>> wrote:



It was working and now it isn't. I think it is an issue with how the sources are setup. I'll give it a look here.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608210307>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITXXA5KGWWAKZD72633RKVKDHANCNFSM4L2XGCHA>.

Besler commented:
@stevenkboyd Can you try again now?

Besler commented:
If it works, we should do two things:
1) Remove `aix` from the bonelab repository
2) Update `AimIO` to version 1.1 and get a new distribution on anaconda.
Agree?

stevenkboyd commented:
I’m about to merge the steve/aix branch to the master (if I can remember how). I added one more output for the memory size. It’s close to the results that Scanco’s method provides, but slightly off. Good enough.

I actually think keeping the python implementation in bonelab would be good. It’s a handy utility, and sometimes we need it out of the context of FEA.

I do think that it’s time to create v9.0 of n88. What do you think? Now that we’re python3 and have made some other changes it’d be nice to make it all official.

Steve

On Apr 2, 2020, at 10:28 PM, Bryce Besler <notifications@github.com> wrote:

﻿

If it works, we should do two things:

  1.  Remove aix from the bonelab repository
  2.  Update AimIO to version 1.1 and get a new distribution on anaconda.
Agree?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Numerics88/AimIO/pull/4#issuecomment-608222283>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AFAQITRFONNMIDAIDQAR6V3RKVQWNANCNFSM4L2XGCHA>.
